### ISSUE TITLE
Table cell dividers

### ISSUE THREAD

**Issue by littleBitsman** [emilk/egui#4783]
**Is your feature request related to a problem? Please describe.** Kinda. I want to make cell dividers but adding separators makes the text *literally disappear*, which is a problem on its own.

**Describe the solution you'd like**
I want the ability to add cell dividers (both between rows and columns) and a border around the edge of the table. All of which should be customizable by color and size in pixels.

**Describe alternatives you've considered**
I tried using separators (e.g., `ui.add(Separator::default().vertical());` on both sides of the label in each row/column) which made the text disappear.

I haven't seen any issues for this so I'm making this one

**Comment by littleBitsman** [comment]
Turns out this is (kinda) possible using `Frame` and `Stroke` but that guarantees a full border (all 4 sides)
```rust
Frame::none().stroke(Stroke::new(2.0, Rgba::from_black_alpha(1.0))).show(ui, |ui| {
    // whatever here...
})
```

**Comment by YgorSouza** [comment]
You can also draw the lines manually with `ui.painter().vline()` and `hline()`, although it is a bit cumbersome to keep track of all the rects. This does however give you complete flexibility to draw any lines you want on top of the table. If it were to get an API for that, it would be a matter of deciding what subset of this functionality it should support, e.g., include or exclude the header, or make shorter lines in the middle instead of going end to end.

**Comment by littleBitsman** [comment]
Yeah, more controllable styles would be really nice especially for things like Tables and Grids. #3284 seems to be getting onto that though.

**Comment by littleBitsman** [comment]
I made these utilities just to make it a bit faster for me to write:
```rs
fn draw_hlines<R>(ui: &mut Ui, length: f32, next: impl FnOnce(&mut Ui) -> R) {
  ui.painter().hline(0.0..=length, 0.0, PathStroke::new(2.0, Rgba::from_rgb(0.0, 0.0, 0.0)));
  next(ui);
  ui.painter().hline(0.0..=length, ui.min_rect().height() + 2.0, PathStroke::new(2.0, Rgba::from_rgb(0.0, 0.0, 0.0)));
}
fn draw_vlines<R>(ui: &mut Ui, height: f32, next: impl FnOnce(&mut Ui) -> R) {
  ui.painter().vline(0.0, 0.0..=height, PathStroke::new(2.0, Rgba::from_rgb(0.0, 0.0, 0.0)));
  next(ui);
  ui.painter().vline(ui.min_rect().width() + 2.0, 0.0..=height, PathStroke::new(2.0, Rgba::from_rgb(0.0, 0.0, 0.0)));
}
fn draw_box<R>(ui: &mut Ui, length: f32, height: f32, next: impl FnOnce(&mut Ui) -> R) {
  draw_hlines(ui, length, |ui| draw_vlines(ui, height, next))
}
```

Doesn't seem like anything is being drawn as a line, unless I'm doing something wrong lol
